src/protocols/secure_channel/CASESession.cpp

```diff
@@ -205,16 +203,14 @@ CHIP_ERROR CASESession::FromSerializable(const CASESessionSerializable & seriali
     VerifyOrReturnError(length <= sizeof(mMessageDigest), CHIP_ERROR_INVALID_ARGUMENT);
     memcpy(mMessageDigest, serializable.mMessageDigest, length);
 
-    length = LittleEndian::HostSwap16(serializable.mIPKLen);
-    VerifyOrReturnError(length <= sizeof(mIPK), CHIP_ERROR_INVALID_ARGUMENT);
-    memcpy(mIPK, serializable.mIPK, length);
-
     SetPeerNodeId(LittleEndian::HostSwap64(serializable.mPeerNodeId));
     SetLocalSessionId(LittleEndian::HostSwap16(serializable.mLocalSessionId));
     SetPeerSessionId(LittleEndian::HostSwap16(serializable.mPeerSessionId));
 
     memcpy(mResumptionId, serializable.mResumptionId, sizeof(mResumptionId));
 
+    memcpy(mIPK, GetIPKList()->data(), sizeof(mIPK));
```

Comment:
What if `GetIPKList()` has a return with the wrong size? It should be checked if possible